### ISSUE TITLE
TypeScript rewrite

### ISSUE THREAD

**Issue by abraham** [PerfectlyNormal/tinymce-rails-imageupload#97]
Any interest in a TypeScript rewrite? The code is a bit of a mess and updating it to modern patterns would clean it up a lot. I've not attempted to compile TS in a gem before but in theory the compiled JS could be generated on publish and committed to git.

**Comment by PerfectlyNormal** [comment]
That would absolutely be interesting. I use a lot of TypeScript daily now, and it is so much nicer to work with. Never wrapped my head around writing definition files and similar though.

Something I've toyed with for a while, but never gotten around to, is to simply drop the whole gem shenanigans. It feels like a bit of a mess to bundle JavaScript in a Ruby gem that gets installed. But it might be required/easier in order to get it to work with TinyMCE. I sadly don't use much TinyMCE or Rails these days, so haven't kept up with the recommended practices

**Comment by abraham** [comment]
I don't do a lot of TinyMCE development but I've got a client that wants some features from time to time.

I'm not all that familiar having TS in a gem but I think for ease of use the compiled JS should probably be committed to git and published with the gem. Users probably shouldn't have to have TS/Node available to use it.

**Comment by PerfectlyNormal** [comment]
Yeah, I agree. It's a beginning, and definitely better than how it is today, so if you feel like working on it, I'll merge it when it's done.
